src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborReaderTests.SkipValue.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Test.Cryptography;
+using Xunit;
+
+namespace System.Security.Cryptography.Encoding.Tests.Cbor
+{
+    public partial class CborReaderTests
+    {
+        [Theory]
+        [MemberData(nameof(SampleValues))]
+        public static void SkipValue_RootValue_HappyPath(string hexEncoding)
+        {
+            byte[] encoding = hexEncoding.HexToByteArray();
+            var reader = new CborReader(encoding);
+
+            reader.SkipValue();
+            Assert.Equal(CborReaderState.Finished, reader.Peek());
+        }
+
+        [Theory]
+        [MemberData(nameof(SampleValues))]
+        public static void SkipValue_NestedValue_HappyPath(string hexEncoding)
+        {
+            byte[] encoding = $"8301{hexEncoding}03".HexToByteArray();
+            var reader = new CborReader(encoding);
+
+            reader.ReadStartArray();
+            reader.ReadInt64();
+            reader.SkipValue();
+            reader.ReadInt64();
+            reader.ReadEndArray();
+            Assert.Equal(CborReaderState.Finished, reader.Peek());
+        }
+
+        [Theory]
+        [MemberData(nameof(SampleValues))]
+        public static void SkipValue_TaggedValue_HappyPath(string hexEncoding)
+        {
+            byte[] encoding = $"c2{hexEncoding}".HexToByteArray();
+            var reader = new CborReader(encoding);
+
+            reader.ReadTag();
+            reader.SkipValue();
+            Assert.Equal(CborReaderState.Finished, reader.Peek());
+        }
+
+        [Fact]
+        public static void SkipValue_NotAtValue_ShouldThrowInvalidOperationException()
+        {
+            byte[] encoding = "80".HexToByteArray();
+            var reader = new CborReader(encoding);
+
+            reader.ReadStartArray();
+            Assert.Throws<InvalidOperationException>(() => reader.SkipValue());
+        }
+
```

Comment:
Please add a test like https://github.com/dotnet/runtime/blob/47ab0dc0c1f4f0a137624c2a7fa1c43414b72f2c/src/libraries/System.Security.Cryptography.Encoding/tests/Asn1/Reader/PeekTests.cs#L142-L177.

(That is, locally roll back to the recursive implementation; figure out where you would have blown out the stack, slap a zero on the end, and show that you can handle anything in the iterative approach.)

Presumably you get to do something simple like

```C#
byte[] encoding = new byte[16384];
encoding.AsSpan().Fill(0xC2);
encoding[encoding.Length - 1] = 0x03;
...
```

Reply:
Done.